checkov/common/multi_signature.py

```diff
@@ -1,16 +1,33 @@
+from __future__ import annotations
+
 import inspect
 from abc import ABCMeta
 from functools import update_wrapper
-from typing import Callable, Any, TypeVar, Dict, List
+from types import CodeType
+from typing import Callable, Any, TypeVar, cast
+from typing_extensions import Protocol
+
+_MultiT = TypeVar("_MultiT")
```

Comment:
🤔

Reply:
here I'm still not really sure, how to type hint this whole file. It is basically a place holder, which tells the type checker, that whatever you put in will come out in different places as the same type